jscomp/others/js_array_buffer.res

```diff
@@ -0,0 +1,6 @@
+type t = {
```

Comment:
I'd suggest marking the record as private so it's not possible to create by hand.

Reply:
Since not every record with the fields is the ArrayBuffer

Reply:
That's basically prevented by nominal typing

Reply:
I mean to prevent the code like this:

```
let myArrayBuffer: Js.ArrayBuffer.t = {
  byteLenght: 2,
  maxByteLength: 4,
  detached: false,
  resizable: false
}
```

Reply:
If we don't want that, we shouldn't use the binding pattern of exposing getters as record fields for every this-bounded object.

I thought it was already a widely used (even encouraged) pattern, more for practicality than strictness. for example: https://github.com/TheSpyder/rescript-webapi/pull/78#issuecomment-1003248560

Reply:
I personally use `private` for my projects. It kind of solves the problem:

```
type t = private {
  byteLength: int,
  maxByteLength: int,
  detached: bool,
  resizable: bool,
}
```

But I see one downside compared to `getter` pattern:

You need to annotate the value when you access the field

```
let fn = (arrayBuffer: ArrayBuffer.t) => arrayBuffer.byteLenght
// vs
let fn = (arrayBuffer) => arrayBuffer->Js.ArrayBuffer.byteLength
```

It's fine for this case but when you have a type with a lot of complicated type arguments I prefer not to write annotations.

Reply:
Maybe we can just leave `type t` here, but what about the Core's direction? @zth

I'll tune in to that.

Reply:
Wait, so does it make sense to provide compatibility with Js' type in the first place?